src/main.rs

```diff
@@ -1,155 +1,128 @@
-use hyper::{Request, Response};
-use hyper::service::service_fn;
-use hyper::server::conn::http1;
-use hyper_util::rt::TokioIo;
+use serde::{de::IntoDeserializer, Deserialize};
 use solana_client::rpc_client::RpcClient;
-use solana_sdk::program_pack::Pack;
-use solana_sdk::pubkey::Pubkey;
+use solana_sdk::{program_pack::Pack, pubkey::Pubkey};
 use spl_token::state::Mint;
 use tokio::net::TcpListener;
 
 lazy_static::lazy_static! {
-    static ref HNT_MINT: Pubkey = Pubkey::from_str_const("hntyVP6YFm1Hg25TN9WGLqM12b8TQmcknKrdu1oxWux");
-    static ref MOBILE_MINT: Pubkey = Pubkey::from_str_const("mb1eu7TzEc71KxDpsmsKoucSSuuoGLv1drys1oP2jh6");
-    static ref IOT_MINT: Pubkey = Pubkey::from_str_const("iotEVVZLEywoTn1QdwNPddxPWszn3zFhEot3MfL9fns");
-
     static ref SOLANA_RPC: String = env::var("SOLANA_RPC").unwrap_or_else(|_| "https://api.mainnet-beta.solana.com".to_string());
 }
 
-const HNT_MAX_SUPPLY: f64 = 223_000_000f64;
-const MOBILE_MAX_SUPPLY: f64 = 230_000_000_000f64;
-const IOT_MAX_SUPPLY: f64 = 200_000_000_000f64;
-
 #[tokio::main]
-pub async fn main() -> Result<(), Box<dyn std::error::Error + Send + Sync>> {
+async fn main() {
     pretty_env_logger::init();
     dotenv().ok();
 
-    let port: u16 = env::var("PORT").unwrap_or_else(|_| "3000".to_string()).parse().expect("PORT must be a number");
+    let port: u16 = env::var("PORT")
+        .unwrap_or_else(|_| "3000".to_string())
+        .parse()
+        .expect("PORT must be a number");
+
+    let app = Router::new()
+        .route("/api/stats/supply/{token}", get(handle_supply))
+        .fallback(async || Redirect::permanent("https://world.helium.com"));
 
     let addr = SocketAddr::from(([0, 0, 0, 0], port));
-    let listener = TcpListener::bind(addr).await?;
+    let listener = TcpListener::bind(addr).await.unwrap();
 
     println!("Listening on http://{}", addr);
 
-    loop {
-        let (stream, _) = listener.accept().await?;
-        let io = TokioIo::new(stream);
-        tokio::task::spawn(async move {
-            if let Err(err) = http1::Builder::new()
-                .serve_connection(io, service_fn(handler))
-                .await
-            {
-                eprintln!("Error serving connection: {:?}", err);
-            }
-        });
-    }
+    axum::serve(listener, app).await.unwrap();
 }
 
-async fn handler(req: Request<hyper::body::Incoming>) -> Result<Response<Full<Bytes>>, Infallible> {
-    let supply_type = maybe_get_type(req.uri().query());
-    match req.uri().path() {
-        "/api/stats/supply/iot" => return handle_supply(&MOBILE_MINT, supply_type),
-        "/api/stats/supply/mobile" => return handle_supply(&IOT_MINT, supply_type),
-        "/api/stats/supply/hnt" => return handle_supply(&HNT_MINT, supply_type),
-        _ => {}
-    }
-
-    // If not a supply API call, fall back to a redirect.
-    redirect_world()
+#[derive(Deserialize, Debug)]
+enum TokenType {
+    #[serde(rename = "hnt")]
+    HNT,
```

Comment:
no need to capitalize.. just use `Hnt` and so on

Reply:
addressed in eb15cab